api/client.go

```diff
@@ -36,14 +36,18 @@ var DefaultRoundTripper http.RoundTripper = &http.Transport{
 	TLSHandshakeTimeout: 10 * time.Second,
 }
 
+type HttpClient interface {
```

Comment:
Why not using simply `http.RoundTripper`?

Reply:
Also if anything this is not following style guide (should be `HTTPClient`) and does not have commentary as public interface

Reply:
It's bit different method name, but still.. `RoundTrip(*Request) (*Response, error)`

In theory the api/http packages are experimental so we can change them with breaking changes.

Reply:
But even with all of this the use case mentioned was possible without this change I believe: 

There is even documentation how to do it here: [github.com/hashicorp/go-retryablehttp/Client](https://github.com/hashicorp/go-retryablehttp?tab=readme-ov-file#getting-a-stdlib-httpclient-with-retries)

Reply:
Hi @bwplotka , thanks for your comments.

To start with, you cannot get HTTP retries just by replacing the `RoundTripper` (I've tried that...), you need an **alternative client**. The retryable client (like hashicorp's) does more stuff than roundtrips, like draining the response body after each retry etc.

Your suggested solution (sorry, somehow I missed that in their docs) to use their `StandardClient()` [converter](https://github.com/hashicorp/go-retryablehttp?tab=readme-ov-file#getting-a-stdlib-httpclient-with-retries) to the library's `http.Client` works - I have just tried that out and saw retries. So the use-case is possible without this change.

In view of that, you can decide. I still don't think that adding the interface is a bad idea as this is what the Prometheus client requires (only the `Do()` function). It's true that hashicorp's client gives you a converter to the standard client, but you are now relying on that. I'll leave it up to you.